ads/openadstream.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {checkData, validateDataExists, writeScript, validateSrcPrefix } from '../src/3p';
+
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function openadstream(global, data) {
+	//all available fields
+	checkData(data, ['url', 'sitepage', 'pos', 'query']);
+	//required fields
+	validateDataExists(data, ['url', 'sitepage', 'pos']);
+	validateSrcPrefix('https:', data.url);
```

Comment:
If `data.url` only contains a hostname, then please do

`'https' + encodeURIComponent(data.hostname)` or something like it instead.